command/agent/agent.go

```diff
@@ -127,28 +130,64 @@ func Create(config *Config, logOutput io.Writer) (*Agent, error) {
 		logOutput = os.Stderr
 	}
 
-	// Validate the config
-	if config.Datacenter == "" {
-		return nil, fmt.Errorf("Must configure a Datacenter")
-	}
-	if config.DataDir == "" && !config.DevMode {
-		return nil, fmt.Errorf("Must configure a DataDir")
-	}
-
-	// Try to get an advertise address
+	var bindIPs []string
```

Comment:
I backed out of a simplification and this was residual.  I just pushed an update that completed this work.